### PR TITLE
Feature/pipeline cli

### PR DESCRIPTION
**Pull request by yarinshaked**

### CODE REVIEW

**Review by eilonshi — APPROVED** [review]

File: pipert2/cli.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import typer
```
Comment (MayoG) [inline pipert2/cli.py:1]: Maybe find a better place for this script, maybe add a new directory in utils or create a new one in the pipert2 folder itself.

Comment (MayoG) [reply pipert2/cli.py:1]: And why do we need this script ?

File: pipert2/core/base/pipe.py [hunk 2]
```diff
@@ -17,49 +20,62 @@
...
 
-    def __init__(self, network: Network = QueueNetwork(),
+    events = class_functions_dictionary()
+
+    def __init__(self, event_board: EventBoard = EventBoard(), network: Network = QueueNetwork(),
                  logger: Logger = get_default_print_logger("Pipe"),
-                 data_transmitter: DataTransmitter = BasicTransmitter(),
-                 auto_pacing_mechanism: bool = False):
+                 data_transmitter: DataTransmitter = BasicTransmitter(), auto_pacing_mechanism: bool = False,
+                 run_rpc_cli: bool = False):
```
Comment (MayoG) [inline pipert2/core/base/pipe.py:36]: I am sad to see that you chose to do this that way. buy i approve :disappointed: . (resolve when you see this)

File: pipert2/core/base/wrappers/rpc_pipe_wrapper.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        self.notify_callback(event_name=STOP_EVENT_NAME)
+
+    def kill(self):
+        """Invokes the kill event in the pipe
+
+        """
+        self.notify_callback(event_name=JOIN_EVENT_NAME, to_kill=True)
+        self.stop()
+
+    def execute(self, name: str, encoded_arguments: str = None):
```
Comment (MayoG) [inline pipert2/core/base/wrappers/rpc_pipe_wrapper.py:48]: Why not getting the arguments in dict instead of string ?

**Review by MayoG — CHANGES_REQUESTED** [review]

**Review by MayoG — APPROVED** [review]

**Pull request closed without merging**